src/orion/algo/space.py

```diff
@@ -377,7 +405,7 @@ def __init__(self, name, categories, **kwargs):
 
         """
         if isinstance(categories, dict):
-            self.categories = tuple(categories.keys())
+            self.categories = tuple(map(lambda k: str(k), categories.keys()))
```

Comment:
We need to make sure the default value is contained in the Categories, that's for sure. I don't see why it would be difficult to do such a test in python. The following works just fine

```python
>>> "aqweqwe" in ["aqweqwe", 0, 1, 1.0]
True
>>> 0 in ["aqweqwe", 0, 1, 1.0]
True
>>> 0.1 in ["aqweqwe", 0, 1, 1.0]
False
>>> numpy.float64(1) in  ["aqweqwe", 0, 1, 1.0]
True
```